### PR TITLE
8143041: Unify G1CollectorPolicy::PauseKind and G1YCType

### PR DESCRIPTION
**Pull request by walulyai**
Please review this refactoring change to unify G1CollectorPolicy::PauseKind and G1YCType enums under the same file. Additionally, make the distinction between collector types and phases clearer.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8143041](https://bugs.openjdk.java.net/browse/JDK-8143041): Unify G1CollectorPolicy::PauseKind and G1YCType

### Reviewers
 * [Thomas Schatzl](https://openjdk.java.net/census#tschatzl) (@tschatzl - **Reviewer**)
 * [Albert Mingkun Yang](https://openjdk.java.net/census#ayang) (@albertnetymk - Committer)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2870/head:pull/2870`
`$ git checkout pull/2870`

### CODE REVIEW

File: src/hotspot/share/gc/g1/g1GCTypes.hpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2012, 2019, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
```
[the remark this answers is not in the document]
Comment (walulyai) [reply src/hotspot/share/gc/g1/g1GCTypes.hpp:2]: webrev shows this as a new file, while github shows it as a renamed file. Not sure which one to consider.

**Comment by walulyai** [comment]
Thanks @tschatzl , I have updated with the proposed changes

**Review by tschatzl — APPROVED** [review]
Lgtm.

**Review by albertnetymk — APPROVED** [review]

**Comment by walulyai** [comment]
Thanks @tschatzl and @albertnetymk for the reviews

**Pull request closed without merging**